### DISCUSSION TITLE
Get sequence ID of process

### DISCUSSION THREAD

**Discussion by dilame** [dokku/dokku#4475] [Q&A] [answered]
Let's pretend I do `dokku ps:scale app worker=2`. Is starts 2 worker containers. Is there a way the first worker can determine it's `1`, and the second is `2`?
I just need to balance jobs between workers, and the best way to do it is to divide `job id` by the workers number with the remainder, but i need to know the worker ID to assign a job.

**Comment by haliliceylan** [comment]
actually, its not a dokku problem. You need to use a worker-queue system for work balancing. if u want any id for workers, you can use system hostname directly. each container has unique name.


or you can use a rabbitmq or queue based message broker, so each worker can get job when it previous job is done. Most of system are running like that, you can look laravel, celery frameworks.

**Comment by dilame** [comment]
> actually, its not a dokku problem. You need to use a worker-queue system for work balancing. if u want any id for workers, you can use system hostname directly. each container has unique name.
> 
> or you can use a rabbitmq or queue based message broker, so each worker can get job when it previous job is done. Most of system are running like that, you can look laravel, celery frameworks.

Thank you for your reply! Your advice is very valuable. I am familiar with the concept of job/message queues. This concept supposes the job to be done. But I got an endless job, it's more like daemons. That's why i need sequential ids for workers.

**Comment by haliliceylan** [comment]
> > actually, its not a dokku problem. You need to use a worker-queue system for work balancing. if u want any id for workers, you can use system hostname directly. each container has unique name.
> > or you can use a rabbitmq or queue based message broker, so each worker can get job when it previous job is done. Most of system are running like that, you can look laravel, celery frameworks.
> 
> Thank you for your reply! Your advice is very valuable. I am familiar with the concept of job/message queues. This concept supposes the job to be done. But I got an endless job, it's more like daemons. That's why i need sequential ids for workers.

maybe you can define id from your side, when a worker connect the your server. you can give a id like dilame-1 dilame-2, and every client with send request with this id, or again maybe you can use container id.

**Comment by josegonzalez** [comment, marked as answer]
There is a `DYNO` environment variable we inject for the docker-local scheduler, but nothing like that for the kubernetes scheduler. The nomad scheduler does that via Nomad itself, though I can't remember the exact environment variable.
